### PR TITLE
Amplify Service check

### PR DESCRIPTION
**Pull request by UnleashedMind**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/lib/amplify-service-auth-check.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const { printAuthErrorMessage } = require('../src/aws-utils/aws-amplify');
+
+async function authDryRun(context, amplifyClient) {
```
Comment (yuth) [inline packages/amplify-provider-awscloudformation/lib/amplify-service-auth-check.js]: can we rename the function to `checkAmplifyServiceIAMPermission`

File: packages/amplify-provider-awscloudformation/lib/attach-backend.js [hunk 2]
```diff
@@ -19,6 +20,13 @@ async function run(context) {
     throw new Error(message);
   }
 
+  const hasPermission = await authDryRun(context, amplifyClient);
+  if (!hasPermission) {
+    const message = 'Permssions to access Amplify service is required.';
+    context.print.error(message);
+    throw new Error(message);
+  }
+
```
Comment (yuth) [inline packages/amplify-provider-awscloudformation/lib/attach-backend.js]: Is this executed when running `amplify pull`?

Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/attach-backend.js]: Yes. 
It is one of the pull workflow when amplify folder is not completely setup for the project.

File: packages/amplify-provider-awscloudformation/lib/amplify-service-auth-check.js (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/amplify-service-auth-check.js]: you got it

**Pull request merged**